usr/sbin/registercloudguest

```diff
@@ -361,36 +393,45 @@ while not base_registered:
                         )
                     )
                     utils.remove_registration_data()
-                    utils.add_hosts_entry(smt)
-                    registration_target = smt
+                    utils.add_hosts_entry(smt_srv)
+                    registration_target = smt_srv
                     break
             break
     else:
         base_registered = True
 
-for product in products:
-    cmd = [
-        register_cmd,
-        '--url',
-        'https://%s' % registration_target.get_FQDN(),
-        '--product',
-        product]
-    if os.path.exists(instance_data_filepath):
-        cmd.append('--instance-data')
-        cmd.append(instance_data_filepath)
-
-    logging.info('Registration: %s' % ' '.join(cmd))
-    p = subprocess.Popen(
-        cmd,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE)
-    res = p.communicate()
-
-    for item in res:
-        entry = item.decode()
-        if 'Error:' in entry:
-            logging.error('\tRegistration failed: %s' % entry)
+base_prod_xml = open('/etc/products.d/baseproduct').read()
+prod_def_start = base_prod_xml.index('<product')
+product_tree = etree.fromstring(base_prod_xml[prod_def_start:])
+prod_identifier = product_tree.find('name').text.lower()
+version = product_tree.find('version').text
+arch = product_tree.find('register').find('target').text.split('-')[-1]
+headers = {'Accept': 'application/vnd.scc.suse.com.v4+json'}
+query_args = 'identifier=%s&version=%s&arch=%s&release_type=ga' % (
```

Comment:
`release_type` isn't an allowed parameter for `/systems/products` endpoint, so it could be dropped (it's ignored by the backend).